File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const res = nums.reduce((acc, item)=> {
            if(acc[item]){
                acc[item]++
            }
            else{
                acc[item] = 1
            }
            return acc
        },{})

        // Step 2 - reverse Sort
        const sorted = Object.entries(res).sort((a,b) => b[1] - a[1])

        // Step 3 - Extract the k elements
        return Object.values(sorted).slice(0,k).map(item => Number(item[0]))


    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int len = nums.length;
        int i = 0;
        int [] output = new int[len];
        while (i < len) {
            int j = 0;
            int result = 1;
            while (j < len) {
                if (i != j) {
                    result = result * nums[j];
                }
                j++;
            }
            output[i] = result;
            i++;
        }
        return output;
    }
}  
